sdk/go/src/sawtooth_sdk/client/signer.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	ellcurv "github.com/btcsuite/btcd/btcec"
+	"math/big"
+)
+
+// -- Key Gen --
+
+// GenPrivKey generates a new private key and returns it as bytes
+func GenPrivKey() []byte {
+	priv, err := ellcurv.NewPrivateKey(ellcurv.S256())
+	if err != nil {
+		panic("Failed to generate private key")
+	}
+	return priv.Serialize()
+}
+
+// GenPubKey generates a new public key from a given private key and returns it
+// using the 33 byte compressed format
+func GenPubKey(privkey []byte) []byte {
+	_, pub := ellcurv.PrivKeyFromBytes(ellcurv.S256(), privkey)
+	return pub.SerializeCompressed()
```

Comment:
Well in the case of public keys, the hex string is the form in which it is used. You would probably speed things up by giving them a hex string so you only encoded it once, not every time a transaction was created.

I chose to also give private keys back in hex string for the JS SDK so it was more consistent, but I could see going either way on that one.